### PR TITLE
Groups: Feature flag validation + analytics property

### PR DESCRIPTION
**Pull request by macobo**
## Changes

This PR adds:
- Group analytics-specific metadata for feature flags
- Validations for feature flags editing if someone sneaks by FE validations

## How did you test this code?

See tests. Also clicked through adding a new flag, didn't run into validation errors.

### CODE REVIEW

File: posthog/api/feature_flag.py [hunk 1]
```diff
@@ -68,6 +69,29 @@ def validate_key(self, value):
...
+                predicate(Property(**property))
+                for condition in filters["groups"]
+                for property in condition.get("properties", [])
+            )
+
+        if aggregation_group_type_index is None:
+            is_valid = properties_all_match(lambda prop: prop.type in ["person", "cohort"])
+            if not is_valid:
+                raise serializers.ValidationError("Filters are not valid (can only use person and cohort properties)")
+        elif aggregation_group_type_index is not None:
```
Comment (hazzadous) [inline posthog/api/feature_flag.py:86]: non-blocking

I guess this is equivalent to just else? To communicate that this is an either or to future developers perhaps just use `else:`?

File: posthog/api/feature_flag.py [hunk 2]
```diff
@@ -68,6 +69,29 @@ def validate_key(self, value):
 
         return value
 
+    def validate_filters(self, filters):
+        aggregation_group_type_index = filters.get("aggregation_group_type_index", None)
```
Comment (hazzadous) [inline posthog/api/feature_flag.py:73]: non-blocking

It mentions that filters isn't required [here](https://github.com/PostHog/posthog/pull/7363/files#diff-901d485ec1a60127f907399dbbf514d89d3a99456190adfa242399d61749d3c5R23), do we need to check that this isn't `None`?

non-blocking as seems like a silly case anyway

File: posthog/models/feature_flag.py [hunk 3]
```diff
@@ -62,6 +62,7 @@ def get_analytics_metadata(self) -> Dict:
             "has_rollout_percentage": any(condition.get("rollout_percentage") for condition in self.conditions),
             "filter_count": filter_count,
             "created_at": self.created_at,
+            "aggregating_by_groups": self.aggregation_group_type_index is not None,
```
Comment (hazzadous) [inline posthog/models/feature_flag.py:65]: I don't see anything referencing `aggregating_by_groups`, should there be a test that depends on this?

**Review by hazzadous — APPROVED** [review]
Just a comment re. `aggregating_by_groups`, it doesn't look like it's used or tested, but that's minor, fine to get in.

File: posthog/api/feature_flag.py (hunk 1 above)
Comment (macobo) [reply posthog/api/feature_flag.py:86]: You're right, it should be `else`. Victim of refactoring :)

File: posthog/api/feature_flag.py (hunk 2 above)
Comment (macobo) [reply posthog/api/feature_flag.py:73]: While the link doesn't work I assume you mean `def filters` in the model.

So this is verifying the data received from FE - which will be consistent. The schema used to be different a long time ago in a way we couldn't easily fix in migrations (now probably could) so the model has a safe-guard which isn't relevant here.

**Pull request merged**